src/python/pants/util/objects.py

```diff
@@ -205,33 +206,33 @@ class EnumVariantSelectionError(TypeCheckError):
   """Raised when an invalid variant for an enum() is constructed or matched against."""
 
 
-def enum(*args):
+def enum(all_values, field_name='value'):
   """A datatype which can take on a finite set of values. This method is experimental and unstable.
 
   Any enum subclass can be constructed with its create() classmethod. This method will use the first
   element of `all_values` as the default value, but enum classes can override this behavior by
   setting `default_value` in the class body.
 
+  If `all_values` contains only strings, then each variant is made into an attribute on the
+  generated enum class object. This allows code such as the following:
+
+  class MyResult(enum(['success', 'failure'])):
+    pass
+
+  MyResult.success # The same as: MyResult(success)
```

Comment:
Should this be:
```suggestion
  MyResult.success # The same as: MyResult('success')
```
and the line below?

Reply:
Yes indeed! Thank you!